src/idl_parser.cpp

```diff
@@ -3466,8 +3443,8 @@ CheckedError Parser::CheckPrivateLeak() {
 CheckedError Parser::CheckPrivatelyLeakedFields(const Definition &def,
                                                 const Definition &value_type) {
   if (!opts.no_leak_private_annotations) return NoError();
-  const auto is_private = def.attributes.Lookup("private");
-  const auto is_field_private = value_type.attributes.Lookup("private");
+  auto *const is_private = def.attributes.Lookup("private");
```

Comment:
This doesn't seem right, its a const pointer to an auto-deduced type, but before it was a const auto-deduced type.

Reply:
Actually, Most of These changes have been applied by clang-tidy, I will check them, BTW, I will limit the checks, and re-run clang-tidy.

Reply:
That's the problem with our over use of `auto` I don't know the type, and its hard to review without an IDE where I can lookup the type easily. 

I manually looked up the type returned and its `Value*` (value is a struct). So now the return type is:

`
Value * * const is_private;
`
Constant pointer to a Value pointer.

instead of
`
const Value* is_private
`

A pointer to a const Value pointer.

Reply:
Actually, it is not `Value * * const is_private;` because compiler deduce it to `Value *const` . Previously it will be also deduced the same way. the new version is just for safety.

Your point is to not use auto ?

Reply:
yeah, `auto` should be for iterators and other types that are very verbose, not common types like `std::string` or user-defined types like `Value` or `StructDef` etc..